tests/labeler/chexpert/test_label.py

```diff
@@ -0,0 +1,313 @@
+from knodle.labeler.CheXpert.neg_unc_detection import NegUncDetector
+from knodle.labeler.CheXpert.updating import Updater
+from knodle.labeler.CheXpert.utils import z_matrix_fct, t_matrix_fct, get_rule_idx
+
+
+@pytest.fixture
+def directories():
+
+    chexpert_data_dir = os.path.join(os.getcwd())
+
+    # RULE DIRECTORIES -----------------------------------------------------------------------------------
+    mention_data_dir = os.path.join(chexpert_data_dir, "phrases", "mention")
+    os.makedirs(mention_data_dir, exist_ok=True)
+    with open(os.path.join(mention_data_dir, 'storm.txt'), 'w') as f:
+        f.write('storm')
+        f.close()
+    with open(os.path.join(mention_data_dir, 'clouds.txt'), 'w') as f:
+        f.write('cloud')
+        f.close()
+    with open(os.path.join(mention_data_dir, 'rain.txt'), 'w') as f:
+        f.write('rain\n')
+        f.write('pour\n')
+        f.write('drizzle')
+        f.close()
+
+    unmention_data_dir = os.path.join(chexpert_data_dir, "phrases", "unmention")
+    os.makedirs(unmention_data_dir, exist_ok=True)
+    with open(os.path.join(unmention_data_dir, 'rain.txt'), 'w') as f:
+        f.write('dry')
+        f.close()
+
+    # PATTERN DIRECTORY ----------------------------------------------------------------------------------
+    patterns_dir = os.path.join(chexpert_data_dir, "patterns")
+    os.makedirs(patterns_dir, exist_ok=True)
+    with open(os.path.join(patterns_dir, 'negation.txt'), 'w') as f:
+        pass
+    with open(os.path.join(patterns_dir, 'pre_negation_uncertainty.txt'), 'w') as f:
+        pass
+    with open(os.path.join(patterns_dir, 'post_negation_uncertainty.txt'), 'w') as f:
+        f.write('{} < {} {lemma:/possible/}')
+        f.close()
+
+    # SAMPLE DIRECTORY -----------------------------------------------------------------------------------
+    sample_dir = os.path.join(chexpert_data_dir, "reports")
+    os.makedirs(sample_dir, exist_ok=True)
+    data = ['The weather is acting up today, even a storm is possible.']
+    with open(os.path.join(sample_dir, 'weather_forecast.csv'), 'w', encoding='UTF8') as f:
+        writer = csv.writer(f)
+        writer.writerow(data)
+
+    # OUTPUT DIRECTORY -----------------------------------------------------------------------------------
+    output_dir = os.path.join(chexpert_data_dir, "output")
+    os.makedirs(output_dir, exist_ok=True)
+
+    return sample_dir, output_dir
+
+
+@pytest.fixture
+def classes():
+
+    preprocessor = Preprocessor(config=WeatherTestsConfig())
+    matcher = Matcher(config=WeatherTestsConfig(), chexpert_data=False)
+    neg_unc_detector = NegUncDetector(config=WeatherTestsConfig())
+    updater = Updater(config=WeatherTestsConfig())
+
+    return preprocessor, matcher, neg_unc_detector, updater
+
+
+# TESTS: "LABEL"--------------------------------------------------------------------------------------
+def test_chexpert_labeler_label(directories):
+
+    directories()
```

Comment:
This would be correct way to use fixtures: https://github.com/knodle/knodle/blob/develop/tests/transformation/test_majority.py#L89

sample_dir, output_dir = directories

Reply:
I also need to run the function as well though, that's why I used directories() all the time. Since otherwise the directories weren't created.